resolve/matrix/Sparse.cpp

```diff
@@ -70,69 +89,132 @@ namespace ReSolve { namespace matrix {
     owns_gpu_data_ = false;
     owns_gpu_vals_ = false;
   }
-
+  
+  /**
+   * @brief destructor
+   * */
   Sparse::~Sparse()
   {
     this->destroyMatrixData(memory::HOST);
     this->destroyMatrixData(memory::DEVICE);
   }
 
+  /** 
+   * @brief set the matrix update flags to false (for both HOST and DEVICE).
+   */
   void Sparse::setNotUpdated()
   {
     h_data_updated_ = false;
     d_data_updated_ = false; 
   }
   
+  /**
+   * @brief get number of matrix rows
+   *
+   * @return number of matrix rows.
+   */
   index_type Sparse::getNumRows()
   {
     return this->n_;
   }
 
+  /**
+   * @brief get number of matrix columns
+   *
+   * @return number of matrix columns.
+   */
   index_type Sparse::getNumColumns()
   {
     return this->m_;
   }
 
+  /**
+   * @brief get number of non-zeros in the matrix.
+   *
+   * @return number of non-zeros.
+   */
   index_type Sparse::getNnz()
   {
     return this->nnz_;
   }
 
+  /**
+   * @brief get number of non-zeros in expanded matrix.
+   *
+   * @return number of non-zeros in expanded matrix.
+   */
   index_type Sparse::getNnzExpanded()
   {
     return this->nnz_expanded_;
   }
 
+  /**
+   * @brief check if matrix is symmetric.
+   *
+   * @return true if symmetric, false otherwise.
+   */
   bool Sparse::symmetric()
   {
     return is_symmetric_;
   }
 
+  /**
+   * @brief check if (symmetric) matrix is expanded.
+   *
+   * @return true if expanded, false otherwise.
+   */
   bool Sparse::expanded()
   {
     return is_expanded_;
   }
 
+  /**
+   * @brief Set matrix symmetry property
+   *
+   * @param[in] symmetric - true to set matrix to symmetric and false to set to non-symmetric 
+   */  
   void Sparse::setSymmetric(bool symmetric)
   {
     this->is_symmetric_ = symmetric;
   }
 
+  /**
+   * @brief Set matrix "expanded" property
+   *
+   * @param[in] expanded - true to set matrix to expanded and false to set to not expanded
+   */  
   void Sparse::setExpanded(bool expanded)
   {
     this->is_expanded_ = expanded;
   }
 
+  /**
+   * @brief Set number of non-zeros in expanded matrix.
+   *
+   * @param[in] nnz_expanded_new - new number of non-zeros in expanded matrix
+   */  
   void Sparse::setNnzExpanded(index_type nnz_expanded_new)
   {
     this->nnz_expanded_ = nnz_expanded_new;
   }
 
+  /**
+   * @brief Set number of non-zeros.
+   *
+   * @param[in] nnz_new - new number of non-zeros
+   */  
   void Sparse::setNnz(index_type nnz_new)
   {
     this->nnz_ = nnz_new;
   }
 
+  /**
+   * @brief Set the data to be updated on HOST or DEVICE. Note it automatically sets the other mirror data to non-updated (but it does not copy).
+   *
+   * @param[in] memspace - memory space (HOST or DEVICE) of data that is set to "updated"
+   *
+   * @return 0 if successful, -1 if not.
+   */  
```

Comment:
Use Doxygen keyword `@note`:
```suggestion
  /**
   * @brief Set the data to be updated on HOST or DEVICE. 
   *
   * @param[in] memspace - memory space (HOST or DEVICE) of data that is set to "updated"
   *
   * @return 0 if successful, -1 if not.
   *
   * @note This method automatically sets the other mirror data to non-updated (but it does not copy).
   */  
```